### DISCUSSION TITLE
Can we add button delete on Edge when hover to edge

### DISCUSSION THREAD

**Discussion by TechGuyVN** [xyflow/xyflow#2347] [Q&A] [answered]
Dear Friend

Can we add button delete on Edge when hover on edge
like this:
<img width="78" alt="image" src="https://user-images.githubusercontent.com/503529/183272800-e94a67e5-13a3-408d-bf19-e8a1ac1c0c0c.png">

Thanks Friends.

**Comment by moklick** [comment, marked as answer]
Hey! Yes, that's possible with a custom edge. We have an example for that: https://reactflow.dev/docs/examples/edges/edge-with-button/

**Reply by TechGuyVN** [reply]
thanks for fast response, how we capture hover event and set visible for button at mouse pointer location ? pls help

**Reply by moklick** [reply]
we have `onEdgeMouseEnter` and `onEdgeMouseLeave` handlers that you could use. I would start with a fixed position for the delete button in the center of the edge.

**Reply by TechGuyVN** [reply]
thanks friend

**Reply by Rugz007** [reply]
This method requires me to implement a global state for my edges. Is there an alternative like passing the props for this? @moklick

**Reply by sleaper** [reply]
@Rugz007 Have you found any solution?

**Reply by ManuTorrado** [reply]
You can use the data object when you update your edges after triggering onEdgeMouseEnter and onEdgeMouseLeave 


```
const [edges, setEdges, onEdgesChange] = useEdgesState([]);
  const onEdgeMouseLeave = (event, edge) => {
    const edgeId = edge.id;

    // Updates edge
    setEdges((prevElements) =>
      prevElements.map((element) =>
        element.id === edgeId
          ? {
              ...element,

              data: {
                ...element.data,

                isHovered: false, 
              },
            }
          : element
      )
    );
  };
```

Then you can use 'isHovered' as a prop on your edge component

**Reply by jazeee** [reply]
Thanks for this. Worked well.
I recommend using `setEdge` from `useReactFlow` (or `setEdgeData`)
https://reactflow.dev/api-reference/hooks/use-react-flow

```
const {  updateEdge } = useReactFlow();
//  this allows you to functionally update a single edge by id:
updateEdge(edgeId, (oldEdge) => ({ data: {...oldEdge.data, isHovered: true });
```
Note:
> This hook can only be used in a component that is a child of a `<ReactFlowProvider />` or a `<ReactFlow />` component.
